frontend/src/components/common/action-button/ActionButton.tsx

```diff
@@ -0,0 +1,45 @@
+import React, { useState } from 'react'
+import cx from 'classnames'
+
+import styles from './ActionButton.module.scss'
+
+const ActionButton = (props: any) => {
+  const { className, children, ...otherProps } = props
+  const [toggleDropdown, setToggleDropdown] = useState(false)
+
+  function handleToggleDropdown(
+    event: React.MouseEvent<HTMLDivElement, MouseEvent>
+  ) {
+    event.stopPropagation()
+    setToggleDropdown(!toggleDropdown)
```

Comment:
Hi @miazima, I think using functional update show clearer intent as the value to update is based on the previous state.
`setToggleDropdown(toggleDropdown => !toggleDropdown)`

reference: https://joaoforja.com/blog/why_use_usestate_functional_update_even_when_everything_seems_to_work/